#150398701 Setup unit testing for front end

Kenec opened the pull request:
#### Set up unit testing for front-end
#### To ensure the front end has all functionality work the way it was intended.
#### To test this run: karma start
#### Setup unit testing for the front end
<img width="1280" alt="screen shot 2017-08-23 at 1 39 17 pm" src="https://user-images.githubusercontent.com/23337808/29616438-67e4404e-8809-11e7-8580-a23572cead91.png">

VanPaitin reviewed (changes requested):
Looks good. Resolve merge conflicts by rebasing and then I will approve. Also note that you will have to get the build to pass

Commit:
chore(karma.config) setup the unit testing for front end

- install the dependencies for the front end test
- write a mock test

[(Delivers) #150398701]

Commit:
fix(karma.config): modified karma conf file

- set the reporter of karma.config file to reporter
- installed dependencies for the reporter

[Fixes #150398701]

Commit:
Merge branch 'staging' of https://github.com/andela/isildur-cfh into staging

fix merge conflict

Kenec commented:
Merge conflict have been resolved by rebasing

Commit:
fix(karma.config): modify karma.config

- add semi colon to resolve hound ci bot issue

[Fixes #150398701]

Kenec commented:
Hound CI bot issue resolved
